machine.go

```diff
@@ -113,23 +125,29 @@ func (c Client) GetMachines(ctx context.Context, p ...MachineListParams) ([]Mach
 	}
 
 	url := fmt.Sprintf("/machines/getMachines")
-	_, err := c.Request(ctx, "GET", url, params, &machines)
+	_, err := c.Request("GET", url, params, &machines, params.RequestParams)
 
 	return machines, err
 }
 
-func (c Client) UpdateMachine(ctx context.Context, p MachineUpdateParams) (Machine, error) {
+func (c Client) UpdateMachine(p MachineUpdateParams) (Machine, error) {
 	machine := Machine{}
 
 	url := fmt.Sprintf("/machines/updateMachine")
-	_, err := c.Request(ctx, "POST", url, p, &machine)
+	_, err := c.Request("POST", url, p, &machine, p.RequestParams)
 
 	return machine, err
 }
 
-func (c Client) DeleteMachine(ctx context.Context, id string) error {
+func (c Client) DeleteMachine(id string, p ...RequestParams) error {
+	var requestParams RequestParams
+
+	if len(p) > 0 {
+		requestParams = p[0]
+	}
+
```

Comment:
Instead of doing this, let's make a consistent parameter style, even for functions that otherwise take a single `id` parameter:
```go
type MachineDeleteParams struct {
	RequestParams
	ID string
}

func (c Client) DeleteMachine(p MachineDeleteParams) error {
	...
```